admin_app/src/app/dashboard/components/performance/ContentsTable.tsx

```diff
@@ -0,0 +1,311 @@
+import { RowDataType } from "@/app/dashboard/types";
+import FilterAltOutlinedIcon from "@mui/icons-material/FilterAltOutlined";
+import SwapVertIcon from "@mui/icons-material/SwapVert";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Pagination from "@mui/material/Pagination";
+import TextField from "@mui/material/TextField";
+import Grid from "@mui/material/Unstable_Grid2";
+import { ApexOptions } from "apexcharts";
+import dynamic from "next/dynamic";
+import { useEffect, useState } from "react";
+
+const ReactApexcharts = dynamic(() => import("react-apexcharts"), {
+  ssr: false,
+});
+
+const QueryCountTimeSeries = ({
+  queryCount,
+  isIncreasing,
+}: {
+  queryCount: number[];
+  isIncreasing: boolean;
+}) => {
+  const series = [
+    {
+      name: "Query Count",
+      data: queryCount,
+    },
+  ];
+  const options: ApexOptions = {
+    chart: {
+      stacked: false,
+      zoom: {
+        enabled: false,
+      },
+      sparkline: {
+        enabled: true,
+      },
+      background: "#fff",
+    },
+    dataLabels: {
+      enabled: false,
+    },
+    stroke: {
+      width: 2,
+      curve: "smooth",
+    },
+    tooltip: {
+      x: {
+        show: false,
+      },
+    },
+    colors: isIncreasing ? ["#4CAF50"] : ["#FF1654"],
+  };
+  return (
+    <ReactApexcharts
+      type="line"
+      options={options}
+      series={series}
+      height={20}
+      width="100%"
+    />
+  );
+};
+
+const SortButton = ({ onClick }: { onClick: () => void }) => {
+  return (
+    <Button
+      size="small"
+      onClick={onClick}
+      sx={{
+        width: "100%",
+        py: 0,
+        fontSize: "small",
+        fontWeight: 300,
+        textTransform: "none",
+        color: "text.secondary",
+        justifyContent: "left",
+        "&:hover": {
+          color: "text.primary",
+          backgroundColor: "transparent",
+          fontWeight: 500,
+        },
+      }}
+    >
+      <SwapVertIcon />
+      Sort
+    </Button>
+  );
+};
+const ContentsTable = ({
+  rows,
+  onClick,
+  rowsPerPage,
+}: {
+  rows: RowDataType[];
+  onClick: () => void;
+  rowsPerPage: number;
+}) => {
+  const [itemsToDisplay, setItemsToDisplay] = useState<RowDataType[]>([]);
+  const [page, setPage] = useState(1);
+  const [sortColumn, setSortColumn] = useState<keyof RowDataType>("query_count");
+  const [sortOrder, setSortOrder] = useState<"ascending" | "descending">("ascending");
+
+  const percentageIncrease = (queryCount: number[]) => {
+    // if the last quarter is greater than the third quarter
+    // then the trend is increasing
+    const queryLength = queryCount.length;
+    const lastQuarterValue = queryCount
+      .slice(Math.floor((queryLength / 4) * 3), queryLength)
+      .reduce((a, b) => a + b, 0);
+    const thirdQuarterValue = queryCount
+      .slice(Math.floor((queryLength / 4) * 2), Math.floor((queryLength / 4) * 3))
+      .reduce((a, b) => a + b, 0);
+
```

Comment:
> The slice() method selects from a given start, up to a (not inclusive) given end.

So we should add +1 to the second arguments.

Also, can we use `round` instead of `floor` and also compare averages instead of summing?

For example, with "Last 24 hours" window we would have 25 points.

The last quarter would be:

- `floor((25/4) * 3) = floor(18.75) = 18`
- `queryCount[18,25+1]` has **8** values at indices 18, 19, 20, 21, 22, 23, 24, 25

The second last quarter would be:

- `floor((25/4) * 2) = floor(12.5) = 12`
- `queryCount[12, 18]` which has **6** values at indices 12, 13, 14, 15, 16, 17

Reply:
It's zero-base like python. So the last index is `queryLength - 1` so the +1 is baked in. 
I'll change it to `round` and also use `average`. Thank you!

Reply:
Im going to fix this in the next PR if that's ok. Save me messing with commits etc.